File: public/PublicMedia/static/solutions/java/Problem7.java
/**
 * Purpose: Find the 10 001st prime number.
 * 
 * @author Eben Quenneville
 * @version 2023-02-10
 */
public class Problem7 {
    // Constants
    private static final int TARGET_PRIME = 10001;

    public static void main(String[] args) {
        int solution = problem7();
        assert solution == 104743;
        System.out.println("Output: " + solution);
    }

    /**
     * 
     * @param n any integer
     * @return whether or not n is a prime number
     */
    private static boolean isPrime(int n) {
        boolean prime = true;
        for (int i = 2; i < (int) (Math.sqrt((double) n)) + 1; i++) {
            if (n % i == 0) {
                prime = false;
                break;
            }
        }
        return prime;
    }

    /**
     * 
     * @return the this.TARGET_PRIMEth prime number, e.g. 10001th prime
     */
    private static int problem7() {
        int candidate = 1, primeCount = 1;
        while (primeCount < TARGET_PRIME) {
            candidate += 2;
            if (isPrime(candidate)) {
                primeCount += 1;
            }
        }
        return candidate;
    }
}

File: public/PublicMedia/static/solutions/rust/problem7.rs
/*
Purpose: Find the 10 001st prime number.

Author: Eben Quenneville
Date: 2023-02-11
*/
const TARGET_PRIME: i32 = 10001;

fn main() {
    let solution = problem7();
    assert!(solution == 104743);
    println!("Output: {}", solution);
}

fn problem7() -> i32 {
    let mut candidate = 1;
    let mut prime_count = 1;
    while prime_count < TARGET_PRIME {
        candidate += 2;
        if is_prime(candidate) {
            prime_count += 1;
        }
    }
    return candidate
}

fn is_prime(n: i32) -> bool {
    let mut prime = true;
    for i in 2..((n as f64).sqrt() as i32) + 1 {
        if n % i == 0 {
            prime = false;
            break;
        }
    }
    return prime
}